pkg/hubble/parser/threefour/parser_test.go

```diff
@@ -1849,17 +1901,22 @@ func TestDecode_TraceNotify(t *testing.T) {
 	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			want := proto.Clone(template)
-			proto.Merge(want, tc.want)
+			isL3Device := strings.HasSuffix(tc.name, "_crypto")
```

Comment:
Feeling slightly uneasy about tying testing logic to the test name
```suggestion
			isL3Device := false
			if ev, ok := tc.event.(monitor.TraceNotifyV1); ok {
				isL3Device = (*monitor.TraceNotify)(&ev).IsL3Device()
			}
```
Granted this suggestion is super ugly, but with the current TraceNotify implementation I don't see how else to do it :sweat_smile:

Reply:
Yeah I was wondering the same, but for sure you're suggestion looks way better than mine :pray: